doc/modules/gaussian_process.rst

```diff
@@ -8,45 +8,514 @@ Gaussian Processes
+which is then squashed through a link function to obtain the probabilistic
+classification. The latent function :math:`f` is a so-called nuisance function,
+whose values are not observed and are not relevant by themselves.
+Its purpose is to allow a convenient formulation of the model, and :math:`f`
+is removed (integrated out) during prediction. GaussianProcessClassifier
+implements the logistic link function, for which the integral cannot be
+computed analytically but is easily approximated in the binary case.
+
+In contrast to the regression setting, the posterior of the latent function
+:math:`f` is not Gaussian even for a GP prior since a Gaussian likelihood is
+inappropriate for discrete class labels. Rather, a non-Gaussian likelihood
+corresponding to the logistic link function (logit) is used.
+GaussianProcessClassifier approximates the non-Gaussian posterior with a
+Gaussian based on the Laplace approximation. More details can be found in
+Chapter 3 of [RW2006]_.
+
+The GP prior mean is assumed to be zero. The prior's
+covariance is specified by a passing a :ref:`kernel <gp_kernels>` object. The
+hyperparameters of the kernel are optimized during fitting of
+GaussianProcessRegressor by maximizing the log-marginal-likelihood (LML) based
+on the passed `optimizer`. As the LML may have multiple local optima, the
+optimizer can be started repeatedly by specifying `n_restarts_optimizer`. The
+first run is always conducted starting from the initial hyperparameter values
+of the kernel; subsequent runs are conducted from hyperparameter values
+that have been chosen randomly from the range of allowed values.
+If the initial hyperparameters should be kept fixed, `None` can be passed as
+optimizer.
+
+:class:`GaussianProcessClassifier` supports multi-class classification
+by performing either one-versus-rest or one-versus-one based training and
+prediction.  In one-versus-rest, one binary Gaussian process classifier is
+fitted for each class, which is trained to separate this class from the rest.
+In "one_vs_one", one binary Gaussian process classifier is fitted for each pair
+of classes, which is trained to separate these two classes. The predictions of
+these binary predictors are combined into multi-class predictions. See the
+section on :ref:`multi-class classification <multiclass>` for more details.
+
+In the case of Gaussian process classification, "one_vs_one" might be
+computationally  cheaper since it has to solve many problems involving only a
+subset of the whole training set rather than fewer problems on the whole
+dataset. Since Gaussian process classification scales cubically with the size
+of the dataset, this might be considerably faster. However, note that
+"one_vs_one" does not support predicting probability estimates but only plain
+predictions. Moreover, note that :class:`GaussianProcessClassifier` does not
+(yet) implement a true multi-class Laplace approximation, but is based on
```

Comment:
Multi-class is achieved by one-vs-rest or one-vs-one and using the binary Laplace approximation for the resulting binary classification tasks. It does not implement the multi-class Laplace approximation, which is presented in Chapter 3.5 of Rasmussen and Williams. The multi-class Laplace would probably be superior but is also fairly involved.